src/plugins/discover/public/application/components/discover_legacy.tsx

```diff
@@ -119,43 +142,30 @@ export function DiscoverLegacy({
   topNavMenu,
   updateQuery,
   updateSavedQueryId,
-}: DiscoverLegacyProps) {
+}: DiscoverProps) {
+  const scrollableDesktop = useRef<HTMLDivElement>(null);
+  const collapseIcon = useRef<HTMLButtonElement>(null);
+  const isMobile = () => {
+    // collapse icon isn't displayed in mobile view, use it to detect which view is displayed
+    return collapseIcon && !collapseIcon.current;
+  };
+
+  const [toggleOn, toggleChart] = useState(true);
   const [isSidebarClosed, setIsSidebarClosed] = useState(false);
-  const { TopNavMenu } = getServices().navigation.ui;
-  const { trackUiMetric } = getServices();
+  const services = getServices();
+  const { TopNavMenu } = services.navigation.ui;
+  const { trackUiMetric } = services;
   const { savedSearch, indexPatternList } = opts;
   const bucketAggConfig = opts.chartAggConfigs?.aggs[1];
   const bucketInterval =
     bucketAggConfig && search.aggs.isDateHistogramBucketAggConfig(bucketAggConfig)
       ? bucketAggConfig.buckets?.getInterval()
       : undefined;
-  const [fixedScrollEl, setFixedScrollEl] = useState<HTMLElement | undefined>();
-
-  useEffect(() => (fixedScrollEl ? opts.fixedScroll(fixedScrollEl) : undefined), [
-    fixedScrollEl,
-    opts,
-  ]);
-  const fixedScrollRef = useCallback(
-    (node: HTMLElement) => {
-      if (node !== null) {
-        setFixedScrollEl(node);
-      }
-    },
-    [setFixedScrollEl]
-  );
-  const sidebarClassName = classNames({
-    closed: isSidebarClosed,
-  });
-
-  const mainSectionClassName = classNames({
-    'col-md-10': !isSidebarClosed,
-    'col-md-12': isSidebarClosed,
-  });
+  const contentCentered = resultState === 'uninitialized';
 
   return (
     <I18nProvider>
-      <div className="dscAppContainer" data-fetch-counter={fetchCounter}>
-        <h1 className="euiScreenReaderOnly">{savedSearch.title}</h1>
+      <EuiPage className="dscPage" data-fetch-counter={fetchCounter}>
```

Comment:
I think we could break down the render function into smaller functions, to improve readability. Eg. `renderSidebar`, `renderRows` etc.

Reply:
If you don't mind we could do that later on ⏳